### PR TITLE
Support pass context.Context to all methods

### PR DESCRIPTION
**Pull request by zchee**
Support pass `context.Context` to all methods.

- [x] is it performance related.

### CODE REVIEW

**Comment by zchee** [comment]
@kanata2 You said
> Actually, we no longer support v1.12, so we should delete webhooks_go112.go

https://github.com/slack-go/slack/pull/1032#pullrequestreview-880975367

So, Remove this file and always use `NewRequestWithContext`.

**Comment by zchee** [comment]
@kanata2 ping

File: chat.go [hunk 1]
```diff
@@ -348,8 +334,8 @@ type responseURLSender struct {
 	deleteOriginal  bool
 }
 
-func (t responseURLSender) BuildRequest() (*http.Request, func(*chatResponseFull) responseParser, error) {
-	req, err := jsonReq(t.endpoint, Msg{
+func (t responseURLSender) BuildRequest(ctx context.Context) (*http.Request, func(*chatResponseFull) responseParser, error) {
```
Comment (kanata2) [inline chat.go]: [NOTE]
`BuildRequest` is public method.
However users cannot use this directly.
So this changes maintain backward compatibility.

File: webhooks.go [hunk 2]
```diff
@@ -31,3 +34,28 @@ func PostWebhookContext(ctx context.Context, url string, msg *WebhookMessage) er
 func PostWebhookCustomHTTP(url string, httpClient *http.Client, msg *WebhookMessage) error {
 	return PostWebhookCustomHTTPContext(context.Background(), url, httpClient, msg)
 }
+
+func PostWebhookContextCustomHTTP(ctx context.Context, url string, httpClient *http.Client, msg *WebhookMessage) error {
+	return PostWebhookCustomHTTPContext(ctx, url, httpClient, msg)
+}
```
Comment (kanata2) [inline webhooks.go]: Why did you add this function?
To me, `PostWebhookCustomHTTPContext` looks good enough 😄

**Review by kanata2 — COMMENTED** [review]
Thanks!

**Comment by zchee** [comment]
@kanata2 Sorry for that, I'd mistake :(

PTAL.

**Review by kanata2 — APPROVED** [review]
LGTM :)

**Comment by zchee** [comment]
@kanata2 I'll merge later.
